### ISSUE TITLE
docker-1.9.0-rc3: port mapping not available in docker ps

### ISSUE THREAD

**Issue by dave-tucker** [docker/libnetwork#716]
Per docker/swarm#1351, I can replicate this issue with plain multi-host networking so re-raising here.

Easily replicated using my setup scripts in dave-tucker/docker-network-demos

```

$ docker $(docker-machine config mhl-demo1) network create -d overlay myapp
c5a05c34821fa1c946e95b7b5377514438b313d8faab4f21745d7c137f17910b

$ docker $(docker-machine config mhl-demo1) run -itd --net=myapp -P nginx
f9e53e995d395b68a292f4a622de1f69930dd27e0e20c27fd3cb8ef3efe0847b

$  docker $(docker-machine config mhl-demo1) ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS               NAMES
f9e53e995d39        nginx               "nginx -g 'daemon off"   7 seconds ago       Up 6 seconds                            prickly_hodgkin

$ docker $(docker-machine config mhl-demo1) inspect f9e53e995d39
[
{
    "Id": "f9e53e995d395b68a292f4a622de1f69930dd27e0e20c27fd3cb8ef3efe0847b",
    "Created": "2015-10-28T20:51:41.430625756Z",
    "Path": "nginx",
    "Args": [
        "-g",
        "daemon off;"
    ],
    "State": {
        "Status": "running",
        "Running": true,
        "Paused": false,
        "Restarting": false,
        "OOMKilled": false,
        "Dead": false,
        "Pid": 3779,
        "ExitCode": 0,
        "Error": "",
        "StartedAt": "2015-10-28T20:51:41.737438059Z",
        "FinishedAt": "0001-01-01T00:00:00Z"
    },
    "Image": "9b922c8dd2d13147addfd9f41c4754bef27659a5d731485a42c728ff5ff9ef34",
    "ResolvConfPath": "/mnt/sda1/var/lib/docker/containers/f9e53e995d395b68a292f4a622de1f69930dd27e0e20c27fd3cb8ef3efe0847b/resolv.conf",
    "HostnamePath": "/mnt/sda1/var/lib/docker/containers/f9e53e995d395b68a292f4a622de1f69930dd27e0e20c27fd3cb8ef3efe0847b/hostname",
    "HostsPath": "/mnt/sda1/var/lib/docker/containers/f9e53e995d395b68a292f4a622de1f69930dd27e0e20c27fd3cb8ef3efe0847b/hosts",
    "LogPath": "/mnt/sda1/var/lib/docker/containers/f9e53e995d395b68a292f4a622de1f69930dd27e0e20c27fd3cb8ef3efe0847b/f9e53e995d395b68a292f4a622de1f69930dd27e0e20c27fd3cb8ef3efe0847b-json.log",
    "Name": "/prickly_hodgkin",
    "RestartCount": 0,
    "Driver": "aufs",
    "ExecDriver": "native-0.2",
    "MountLabel": "",
    "ProcessLabel": "",
    "AppArmorProfile": "",
    "ExecIDs": null,
    "HostConfig": {
        "Binds": null,
        "ContainerIDFile": "",
        "LxcConf": [],
        "Memory": 0,
        "MemoryReservation": 0,
        "MemorySwap": 0,
        "KernelMemory": 0,
        "CpuShares": 0,
        "CpuPeriod": 0,
        "CpusetCpus": "",
        "CpusetMems": "",
        "CpuQuota": 0,
        "BlkioWeight": 0,
        "OomKillDisable": false,
        "MemorySwappiness": -1,
        "Privileged": false,
        "PortBindings": {},
        "Links": null,
        "PublishAllPorts": true,
        "Dns": null,
        "DnsOptions": null,
        "DnsSearch": null,
        "ExtraHosts": null,
        "VolumesFrom": null,
        "Devices": [],
        "NetworkMode": "myapp",
        "IpcMode": "",
        "PidMode": "",
        "UTSMode": "",
        "CapAdd": null,
        "CapDrop": null,
        "GroupAdd": null,
        "RestartPolicy": {
            "Name": "no",
            "MaximumRetryCount": 0
        },
        "SecurityOpt": null,
        "ReadonlyRootfs": false,
        "Ulimits": null,
        "LogConfig": {
            "Type": "json-file",
            "Config": {}
        },
        "CgroupParent": "",
        "ConsoleSize": [
            0,
            0
        ],
        "VolumeDriver": ""
    },
    "GraphDriver": {
        "Name": "aufs",
        "Data": null
    },
    "Mounts": [
        {
            "Name": "b26f041f0f1f6243e0d4aba916f52810eff2224aabb286428328ea6d482f9c54",
            "Source": "/mnt/sda1/var/lib/docker/volumes/b26f041f0f1f6243e0d4aba916f52810eff2224aabb286428328ea6d482f9c54/_data",
            "Destination": "/var/cache/nginx",
            "Driver": "local",
            "Mode": "",
            "RW": true
        }
    ],
    "Config": {
        "Hostname": "f9e53e995d39",
        "Domainname": "",
        "User": "",
        "AttachStdin": false,
        "AttachStdout": false,
        "AttachStderr": false,
        "ExposedPorts": {
            "443/tcp": {},
            "80/tcp": {}
        },
        "Tty": true,
        "OpenStdin": true,
        "StdinOnce": false,
        "Env": [
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "NGINX_VERSION=1.9.5-1~jessie"
        ],
        "Cmd": [
            "nginx",
            "-g",
            "daemon off;"
        ],
        "Image": "nginx",
        "Volumes": {
            "/var/cache/nginx": {}
        },
        "WorkingDir": "",
        "Entrypoint": null,
        "OnBuild": null,
        "Labels": {},
        "StopSignal": "SIGTERM"
    },
    "NetworkSettings": {
        "Bridge": "",
        "SandboxID": "5ab683116a6df1fdfe665f67a8a251e38825a0c093aeec97b42763cbf4fc1427",
        "HairpinMode": false,
        "LinkLocalIPv6Address": "",
        "LinkLocalIPv6PrefixLen": 0,
        "Ports": {},
        "SandboxKey": "/var/run/docker/netns/5ab683116a6d",
        "SecondaryIPAddresses": null,
        "SecondaryIPv6Addresses": null,
        "Networks": {
            "myapp": {
                "EndpointID": "0d84acc3ac3585063f33ec7ccb4a426480ae1673b7435df7bffbb00849562856",
                "Gateway": "\u003cnil\u003e",
                "IPAddress": "10.0.0.2",
                "IPPrefixLen": 24,
                "IPv6Gateway": "",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "MacAddress": ""
            }
        }
    }
}
]
```

There is no port mapping in NetworkSettings but it does appear to be mapped:

```
Proto Recv-Q Send-Q Local Address           Foreign Address         State
tcp        0      0 0.0.0.0:ssh             0.0.0.0:*               LISTEN
tcp        0      0 192.168.99.105:7946     0.0.0.0:*               LISTEN
tcp        0      0 :::www                  :::*                    LISTEN
tcp        0      0 :::ssh                  :::*                    LISTEN
tcp        0      0 :::32770                :::*                    LISTEN
tcp        0      0 :::32771                :::*                    LISTEN
```

**Comment by tomdee** [comment]
I've just hit this too. Shouldn't this issue be on docker/docker? Otherwise it won't be tracked or make it into 1.9?

**Comment by abronan** [comment]
Can confirm as well. Hitting the same issue testing with master on a Digital Ocean cluster :)

**Comment by aboch** [comment]
Closed by docker/docker [#17490](https://github.com/docker/docker/pull/17490)
